### PR TITLE
fix(pt): use user seed in `DpLoaderSet`

### PR DESCRIPTION
**Pull request by iProzd**
## Summary by CodeRabbit

- **New Features**
	- Introduced a `seed` parameter for enhanced control over randomness in the training process.
	- Improved randomness handling for distributed training setups to ensure unique batch sequences.

- **Bug Fixes**
	- Enhanced robustness in seed setup during initialization to prevent unnecessary function calls with a `None` value.

- **Chores**
	- Streamlined the random seed management by removing redundant seed-setting logic in training processes.

### CODE REVIEW

File: deepmd/pt/utils/dataloader.py [hunk 1]
```diff
@@ -80,7 +80,8 @@ def __init__(
         seed=10,
```
Comment (njzjz) [inline deepmd/pt/utils/dataloader.py:80]: Should we change 10 to None?

Comment (njzjz) [reply deepmd/pt/utils/dataloader.py:80]: Or remove the default value

Comment (iProzd) [reply deepmd/pt/utils/dataloader.py:80]: Yes I think None is better.

**Review by CaRoLZhangxy — APPROVED** [review]

**Review by njzjz — APPROVED** [review]

**Review by wanghan-iapcm — APPROVED** [review]
